### PR TITLE
Fix `date_bin` fn someways (like a cat?)

### PR DESCRIPTION
**Pull request by workingjubilee**
This neither is used internally nor is it publicly available, due to invalid cfg attributes on it. Fixing the cfg on it breaks compilation, so the code inside it isn't valid either.

### CODE REVIEW

File: pgrx/src/datum/datetime_support/ctor.rs [hunk 1]
```diff
@@ -85,45 +85,3 @@ pub fn to_timestamp(epoch_seconds: f64) -> TimestampWithTimeZone {
...
-/// ```
-#[cfg(any(features = "pg14", features = "pg15"))]
-pub fn date_bin(
-    stride: crate::datum::interval::Interval,
-    source: Timestamp,
-    origin: Timestamp,
-) -> Timestamp {
-    unsafe {
-        direct_function_call(
-            pg_sys::date_bin,
```
Comment (eeeebbbbrrrr) [inline pgrx/src/datum/datetime_support/ctor.rs:124]: haha.  This should have been `timestamp_bin` and/or `timestamptz_bin`

per Discord chat, we can instead add a `bin(stride, origin)` function to those two types instead.

**Review by eeeebbbbrrrr — CHANGES_REQUESTED** [review]
I'll take over this PR and do something different per the comment over there on line 124

File: pgrx/src/datum/datetime_support/ctor.rs [hunk 2]
```diff
@@ -85,45 +85,3 @@ pub fn to_timestamp(epoch_seconds: f64) -> TimestampWithTimeZone {
...
-/// # Examples
-///
-/// ```sql
-/// SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01');
-/// Result: 2020-02-11 15:30:00
-///
-/// SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01 00:02:30');
-/// Result: 2020-02-11 15:32:30
-/// ```
-#[cfg(any(features = "pg14", features = "pg15"))]
```
Comment (eeeebbbbrrrr) [inline pgrx/src/datum/datetime_support/ctor.rs:116]: s/feature**s**/feature/g

😵‍💫 🌴

Comment (nyurik) [reply pgrx/src/datum/datetime_support/ctor.rs:116]: note that `cargo clippy` catches these - another reason to add clippy to CI ;)

Comment (workingjubilee) [reply pgrx/src/datum/datetime_support/ctor.rs:116]: No.

Comment (workingjubilee) [reply pgrx/src/datum/datetime_support/ctor.rs:116]: Please examine our linting checks as they currently stand: https://github.com/pgcentralfoundation/pgrx/blob/395c97584d1401614d6e0bfa94efa5a1048a199b/.github/workflows/tests.yml#L353-L376

**Comment by workingjubilee** [comment]
Nightly adds a lint that warns on this, and it's very annoying for building pgrx's code with nightly. Sorry, I need to go ahead and fix this.

**Pull request closed without merging**